libcodechecker/analyze/analyzers/analyzer_types.py

```diff
@@ -337,7 +351,24 @@ def __build_clang_tidy_config_handler(args, context):
 
     config_handler = config_handler_clang_tidy.ClangTidyConfigHandler()
     config_handler.analyzer_binary = context.analyzer_binaries.get(CLANG_TIDY)
-    config_handler.compiler_resource_dir = context.compiler_resource_dir
+
+    # FIXME We cannot get the resource dir from the clang-tidy binary,
+    # therefore now we get a clang binary which is a sibling of the clang-tidy.
+    # TODO Support "clang-tidy -print-resource-dir" .
+    check_env = analyzer_env.get_check_env(context.path_env_extra,
+                                           context.ld_lib_path_extra)
+    # Overwrite PATH to contain only the parent of the clang binary.
+    if os.path.isabs(config_handler.analyzer_binary):
+        check_env['PATH'] = os.path.dirname(config_handler.analyzer_binary)
```

Comment:
Is it a good idea to overwrite the PATH? What is the motivation behind this?

Reply:
Why not just extend it?

Reply:
Or in case the point is to always resolve to the absolute clang binary, maybe we could just set the binary instead of invoking the function that looks up the binary in the path?

Reply:
The idea is to lookup `clang` from $PATH when it is a relative path. This is handled with the local object on the stacked with the name `check_env`.
However, when `clang-tidy` is absolute we overwrite the PATH in `check_env` so we lookup only in the parent dir of `clang-tidy`. 

It might be confusing to append to the path, because we may end up with /a/b/c/clang for /x/y/z/clang-tidy. This is true nevertheless in the relative case too, but that we cannot work around.

Reply:
I think if we want tidy and clang from the same directory we should modify the lookup function. 
Overwriting the path might have negative consequences when we use this environment. (E.g.: we can not even issue a `cd` command if we pass this environment to a Popen or similar functions.)

Reply:
Simply setting the binary will miss the binary e.g. `clang-4.0`, this the reason to use `resolve_missing_binary`.

Reply:
We don't write over the effective PATH in the global env. We use a local variable and we overwrite the path in that local variable.